### PR TITLE
add F# templates to `dotnet new`

### PR DESCRIPTION
**Pull request by enricosada**
Added f# templates for `dotnet new`
Same templates as c# (`console`,`lib`,`web`,`xunittest`,`mstest`)

### CODE REVIEW

**Comment by piotrpMSFT** [comment]
@mlorbetske fyi

**Comment by enricosada** [comment]
added tests

**Comment by enricosada** [comment]
the error:

```
18:03:21    System.IO.IOException : The process cannot access the file '\\?\D:\j\workspace\debug_windows---54dcb9ec\test\dotnet-new.Tests\bin\Debug\netcoreapp1.0\When_dotnet_build_is_invoked_Then_app_builds_without_warnings' because it is being used by another process.
```

is because the test dir is created with test name? changed to fix that

**Comment by enricosada** [comment]
ok the test was failing because it was doing (to check two consecutive `dotnet new` should fail)

- `dotnet new -l fsharp`
- `dotnet new` expected to fail, but instead is success because csproj doesnt exists so is created

changed to do create two f# projects:

- `dotnet new -l fsharp`
- `dotnet new -l fsharp` now fails correctly

**Comment by enricosada** [comment]
@piotrpMSFT i think now is ok, i should have fixed the tests (but probably cannot wait to see if ok).

**Comment by enricosada** [comment]
@livarcocc i see you are updating the `web` template, can be done also for f# after this is merged? i see you are near signoff for release of next preview, someone should review this one?

**Comment by enricosada** [comment]
debian test error: `Microsoft.DotNet.Tests.PackagedCommandTests.TestProjectToolIsAvailableThroughDriver` i dont think is related

**Comment by enricosada** [comment]
@dotnet-bot Test Debian8.2 x64 Debug Build

**Comment by enricosada** [comment]
@piotrpMSFT  now finally the pr is green, with test passing for f# 

dont know if is right to do that (or rude) but:

@livarcocc @krwq @MattGertz @srivatsn @mlorbetske for signoff for new release, if possibile? asking for a `approved for next preview` label 😄 

i know some package version are not final (for sure xunit beta3 -> beta4 and `web`). also i see #4821 too with some changes. i can add these changes tomorrow morning for me (9-10h from now) if needed

**Comment by cartermp** [comment]
LGTM 👍

File: test/dotnet-new.Tests/GivenThatIWantANewFSApp.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        public void When_NewtonsoftJson_dependency_added_Then_project_restores_and_runs()
+        {
+            var rootPath = TestAssetsManager.CreateTestDirectory().Path;
+            var projectName = new DirectoryInfo(rootPath).Name;
+            var projectFile = Path.Combine(rootPath, $"{projectName}.csproj");
+
+            new TestCommand("dotnet") { WorkingDirectory = rootPath }
+                .Execute("new --lang f#");
+            
+            GivenThatIWantANewCSApp.AddProjectDependency(projectFile, "Newtonsoft.Json", "7.0.1");
```
Comment (piotrpMSFT) [inline test/dotnet-new.Tests/GivenThatIWantANewFSApp.cs]: This needs to be extracted to another class. The cross-dependency F# on C# is not right...

**Comment by davidfowl** [comment]
@piotrpMSFT @cartermp Do we need to update to use the new attribute syntax?

**Comment by piotrpMSFT** [comment]
We do, but that's a separate PR. @enricosada has been working on adding back F# support for some time and the new syntax was not in the original scope of this PR. @enricosada is that something you can pick up?

**Comment by enricosada** [comment]
@davidfowl @piotrpMSFT working on that now, i just tried to push something out asap so at least works 😄 . i'll send another pr aligned with new web and console/lib templates asap

**Comment by enricosada** [comment]
@davidfowl @piotrpMSFT i sent pr to update all f# templates (web too) with #4836 ( ref https://github.com/dotnet/cli/pull/4836#issuecomment-262821590 )

**Pull request merged**